src/pyodide/internal/python.js

```diff
@@ -69,6 +70,22 @@ import stdlib from "pyodide-internal:generated/python_stdlib.zip";
  */
 let MEMORY = undefined;
 
+/**
+ * Used for tracing via Jaeger.
+ */
+
+export async function enterJaegerSpan(span, callback) {
+  console.log(internalJaeger.traceId);
+  if (!internalJaeger.traceId) {
+    // Jaeger tracing not enabled or traceId is not present in request.
+    return await callback();
+  }
+
+  return await internalJaeger.enterSpan(span, async (span) => {
+    return await callback();
+  });
```

Comment:
any reason not to do:
```suggestion
  return await internalJaeger.enterSpan(span, callback);
```

Reply:
I wanted to make sure the callback is awaited inside the span scope, I was worried that `enterSpan` may not await the callback for me. But I see in its code that it does so I'll change this.